### PR TITLE
Periodic Reports | Skipped Boolean & Reason

### LINKED ISSUE #1998: Periodic Reports | Skipped Boolean & Reason

### PR DESCRIPTION
**Pull request by daniel-leod**
API part of #1998

### CODE REVIEW

File: src/components/periodic-report/dto/periodic-report.dto.ts [hunk 1]
```diff
@@ -53,6 +53,12 @@ class PeriodicReport extends Resource {
   @Field()
   readonly receivedDate: SecuredDateNullable;
 
+  @Field()
+  readonly skipped: boolean;
+
+  @Field()
+  readonly skippedReason?: string;
```
Comment (CarsonF) [inline src/components/periodic-report/dto/periodic-report.dto.ts:60]: These should be secured, and unfortunately that means right now that all the role files need to be updated to allow these two new props.

**Review by CarsonF — CHANGES_REQUESTED** [review]
- [ ] Please add a migration to add these two new props to all existing reports. See [AddPresetInventory](https://github.com/SeedCompany/cord-api-v3/blob/narrative-report-form/src/components/project/migrations/AddPresetInventory.migration.ts) as an example.
- [ ] Secure them

@sethmcknight Should skipped reason be required when skipped is being set to true?

**Comment by sethmcknight** [comment]
> * [ ]  Please add a migration to add these two new props to all existing reports. See [AddPresetInventory](https://github.com/SeedCompany/cord-api-v3/blob/narrative-report-form/src/components/project/migrations/AddPresetInventory.migration.ts) as an example.
> * [ ]  Secure them
> 
> @sethmcknight Should skipped reason be required when skipped is being set to true?

@CarsonF @daniel-leod yes, we should required a reason.

**Comment by CarsonF** [comment]
Since it's required, let's just drop the `skipped` boolean. It can be easily calculated from `skippedReason != null`

File: src/components/periodic-report/periodic-report.repository.ts [hunk 2]
```diff
@@ -39,6 +39,8 @@ export class PeriodicReportRepository extends DtoRepository(IPeriodicReport) {
       type: input.type,
       start: input.start,
       end: input.end,
+      skipped: input.skippedReason != null,
```
Comment (CarsonF) [inline src/components/periodic-report/periodic-report.repository.ts:42]: Err sorry I mean that in general. We don't have to write this to the DB as we can check it on read.

**Review by CarsonF — CHANGES_REQUESTED** [review]
Needs migration still too

**Review by CarsonF — APPROVED** [review]
Thanks @daniel-leod!

**Pull request merged**